x/oracle/keeper/test_common.go

```diff
@@ -52,16 +68,76 @@ func CreateTestKeepers(t *testing.T, isCheckTx bool, consensusNeeded float64, in
 		bank.DefaultCodespace,
 	)
 
+	stakingKeeper := staking.NewKeeper(cdc, keyStaking, tkeyStaking, ck, pk.Subspace(staking.DefaultParamspace), staking.DefaultCodespace)
+	stakingKeeper.SetPool(ctx, staking.InitialPool())
+	stakingKeeper.SetParams(ctx, staking.DefaultParams())
+
 	keeper, keeperErr := NewKeeper(ck, keyOracle, cdc, types.DefaultCodespace, consensusNeeded)
 
+	// create the validators desired and fill them with the expected amount of coins
+	validatorAddresses := createTestAddrs(len(validatorPowers))
+	for index, address := range validatorAddresses {
+		power := validatorPowers[index]
+		coins := cmtypes.TokensFromTendermintPower(power)
+
+		pool := stakingKeeper.GetPool(ctx)
+		_, _, err := ck.AddCoins(ctx, address, sdk.Coins{
+			{stakingKeeper.BondDenom(ctx), coins},
+		})
+		require.Nil(t, err)
+		pool.NotBondedTokens = pool.NotBondedTokens.Add(coins)
+		stakingKeeper.SetPool(ctx, pool)
+	}
+
 	if len(initialClaims) != 0 {
 		for _, claim := range initialClaims {
 			_, err := keeper.ProcessClaim(ctx, claim)
 			require.NoError(t, err)
 		}
 	}
 
-	return ctx, accountKeeper, keeper, keeperErr
+	return ctx, accountKeeper, keeper, validatorAddresses, keeperErr
+}
+
+// nolint: unparam
+func createTestAddrs(numAddrs int) []sdk.AccAddress {
+	var addresses []sdk.AccAddress
+	var buffer bytes.Buffer
+
+	// start at 100 so we can make up to 999 test addresses with valid test addresses
+	for i := 100; i < (numAddrs + 100); i++ {
+		numString := strconv.Itoa(i)
+		buffer.WriteString("A58856F0FD53BF058B4909A21AEC019107BA6") //base address string
+
+		buffer.WriteString(numString) //adding on final two digits to make addresses unique
+		res, _ := sdk.AccAddressFromHex(buffer.String())
+		bech := res.String()
+		addresses = append(addresses, TestAddr(buffer.String(), bech))
+		buffer.Reset()
+	}
+	return addresses
+}
+
+func TestAddr(addr string, bech string) sdk.AccAddress {
+
+	res, err := sdk.AccAddressFromHex(addr)
+	if err != nil {
+		panic(err)
+	}
+	bechexpected := res.String()
+	if bech != bechexpected {
+		panic("Bech encoding doesn't match reference")
+	}
+
+	bechres, err := sdk.AccAddressFromBech32(bech)
+	if err != nil {
+		panic(err)
+	}
+	if bytes.Compare(bechres, res) != 0 {
+		panic("Bech decode and hex decode don't match")
+	}
+
+	return res
 }
 
 // MakeTestCodec creates a codec used only for testing

```

Comment:
Testing Codec was primarily introduced for simplification purposes?

Reply:
Yeh, follows same way done in other cosmos modules